src/parser/WordsAndDocsFileParser.h

```diff
@@ -96,12 +98,40 @@ struct DocsFileLine {
 // Custom delimiter class for tokenization of literals using `absl::StrSplit`.
 // The `Find` function returns the next delimiter in `text` after the given
 // `pos` or an empty substring if there is no next delimiter.
+// This version properly handles Unicode characters using ICU.
 struct LiteralsTokenizationDelimiter {
-  absl::string_view Find(absl::string_view text, size_t pos) const {
+  absl::string_view Find(absl::string_view text, size_t unsignedPos) const {
+    auto isAscii = [](char c) { return static_cast<unsigned char>(c) < 128; };
     auto isWordChar = [](char c) -> bool { return std::isalnum(c); };
-    auto found = std::find_if_not(text.begin() + pos, text.end(), isWordChar);
-    if (found == text.end()) return text.substr(text.size());
-    return {found, found + 1};
+    auto isAsciiWordChar = [isAscii, isWordChar](char c) {
+      return isAscii(c) && isWordChar(c);
+    };
+    // Found non ascii characters.
+    auto pos = static_cast<int64_t>(unsignedPos);
+    auto size = static_cast<int64_t>(text.size());
+    while (pos < size) {
+      // First a fast path for ascii characters.
+      auto found =
+          std::find_if_not(text.begin() + pos, text.end(), isAsciiWordChar);
+      if (found == text.end()) {
+        return {text.end(), text.end()};
+      }
+      if (isAscii(*found)) {
+        return {found, found + 1};
+      }
```

Comment:
Did you check how big the performance impact is of this optimization. From what I saw the only caller of this function does a to-lowercase-transformation after figuring out the delimiters on the individual parts, so at the end of the day we still need to lookup the mapping in the unicode DB, so the performance impact might not be that big.

Reply:
As for now the biggest gains here can be obtained by parallelization first, I will remove the fast path, as it leads to more complex code ("only" relevant during the text index builder which is currently slow anyway.